From 4854ab3f2867aaafee772db3fa8728e5aa4f592c Mon Sep 17 00:00:00 2001
Message-Id: <4854ab3f2867aaafee772db3fa8728e5aa4f592c.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 4 Feb 2015 22:46:42 +0530
Subject: [PATCH 93/95] Revert "REVERTME [VPG]: drm/i915: Skip wait for Gfx,
 if Gfx clock is already On"

Reverting below commit and following patch will have an upstream
implementation.

The Orginal issue was introduced in:
  commit 0003b192f94bba129ae51c302ca0c7f52815643a
  Author: Deepak S <deepak.s@intel.com>
  Date:   Sat Sep 27 14:58:20 2014 +0530

    REVERTME [VPG]: drm/i915: Skip wait for Gfx, if Gfx clock is already
    On

For: GMIN-3623
Change-Id: I2b0013ec12c1aeedcad6d4176557b576edd30229
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 8891f5a..3fb014a 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1694,17 +1694,11 @@ int vlv_force_gfx_clock(struct drm_i915_private *dev_priv, bool force_on)
 	int err;
 
 	val = I915_READ(VLV_GTLC_SURVIVABILITY_REG);
-	/* WARN_ON only for the Valleyview */
-	WARN_ON((!!(val & VLV_GFX_CLK_FORCE_ON_BIT) == force_on) &&
-			!IS_CHERRYVIEW(dev_priv->dev));
+	WARN_ON(!!(val & VLV_GFX_CLK_FORCE_ON_BIT) == force_on);
 
 #define COND (I915_READ(VLV_GTLC_SURVIVABILITY_REG) & VLV_GFX_CLK_STATUS_BIT)
 	/* Wait for a previous force-off to settle */
 	if (force_on) {
-		/* FIXME: On CHV: GFX clock is ON by default. */
-		if (IS_CHERRYVIEW(dev_priv->dev))
-			return 0;
-
 		err = wait_for(!COND, 20);
 		if (err) {
 			DRM_ERROR("timeout waiting for GFX clock force-off (%08x)\n",
-- 
1.7.9.5

